### DISCUSSION TITLE
Query followers/following

### DISCUSSION THREAD

**Discussion by senolatac** [sbertix/Swiftagram#111] [Q&A] [answered]
In swiftagram, you have endpoint to fetch user follower/following list page by page. But I wonder how can I search follower/following users? Is there an endpoint for it?

For Example:

v1/user/friendship/follower?q=abc
Show only my followers, username/name includes "abc"

**Comment by sbertix** [comment]
Hey @senolatac,

No, there isn't a method querying followers/following by name/username, right now. 

It's mostly cause I forgot to add it back 😅

If it's still working like it should I'll push an update within a couple hours.

**Comment by sbertix** [comment, marked as answer]
As of `4.2.4`, `Endpoint.Friendship.followed` and `Endpoint.Friendship.following` allow for an optional parameter (default to `nil` — reverting to the old _no query_ behavior) to query for specific names/usernames. 

```swift
// This works exactly like writing "instagram" on the search
// bar on the followers section of the logged in user.
Endpoint.Friendship.followed(by: secret.id, matching: "instagram")
   .unlocking(with: secret)
   .task(maxLength: 1,
         onComplete: { _ in },
         onChange: { _ in  
            // Do something here.
    })
   .resume()
```
